### PR TITLE
Add community health files

### PR DESCRIPTION
**Pull request by U8NWXD**
* Add code of conduct
* Switch to Apache license and use CLA. The CLA is referenced in a PR template so that every new PR that gets opened will, by default, include a note agreeing to the CLA. I also noted in the contribution guidelines that opening a PR constitutes acceptance of the CLA.
* Add contributing guidelines
* Rearrange the README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,29 +1,64 @@
...
+* (**b**) *Stores* hold the state variables and map each process'
+  variable *names* to their *values*.  They have a schema that
+  determines how the variables are handled with properties such as
+  *units*, *updaters*, *dividers*, *emitters*, and more.
+* (**c**) *Topology* is a bipartite graph of processes connected to
+  stores through their ports.  Shared stores aggregate the processes'
+  required variables, and couple the processes as they unfold in time.
+* (**d**) Processes and Stores can be linked together with a topology in
+  a single level called a compartment, and across compartments by way of
+  boundary stores.
```
Comment (1fish2) [inline README.md:29]: Suggestion: A topology describes how Processes are linked within a Compartment via Stores and how Processes are linked across Compartments via boundary Stores.

File: README.md [hunk 2]
```diff
@@ -1,29 +1,64 @@
...
+* (**e**) Compartments are embedded within each other in a hierarchy --
+  depicted here as a place graph with outer compartments at the top and
+  inner compartments below them.
```
Comment (1fish2) [inline README.md:32]: Suggestion: top/below --> above/below.

File: README.md [hunk 3]
```diff
@@ -1,29 +1,64 @@
...
-![vivarium](doc/_static/interface.png)
+If you want to contribute to Vivarium Core, see our [contribution
+guidelines](CONTRIBUTING.md).
+
+## License
+
+Copyright (C) 2019-2022 The Vivarium Authors
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may
+not use this file except in compliance with the License. You may obtain
```
Comment (1fish2) [inline README.md:54]: By "this file" do you mean this project?

File: CODE_OF_CONDUCT.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+standards, including sustained inappropriate behavior, harassment of an
+individual, or aggression toward or disparagement of classes of
+individuals.
+
+**Consequence**: A permanent ban from any sort of public interaction
+within the community.
+
+## Attribution
+
+This Code of Conduct is adapted from the [Contributor Covenant][homepage],
```
Comment (1fish2) [inline CODE_OF_CONDUCT.md:122]: How does this syntax work? It looks like a cross between a link and a footnote.

**Review by 1fish2 — APPROVED** [review]
Thanks, Chris!

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<!-- Here you should describe what this PR does and why. -->
+
+<!-- DO NOT MODIFY ANYTHING BELOW THIS LINE -->
+-----------------------------------------------------------------------
+
+By creating this pull request, I agree to the Contributor License
```
Comment (eagmon) [inline .github/pull_request_template.md:6]: this is convenient!

File: CODE_OF_CONDUCT.md [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+applies when an individual is officially representing the community in
+public spaces. Examples of representing our community include using an
+official e-mail address, posting via an official social media account,
+or acting as an appointed representative at an online or offline event.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may
+be reported to the Vivarium Authors at the email addresses in
+[`AUTHORS.md`](AUTHORS.md). All complaints will be reviewed and
```
Comment (eagmon) [inline CODE_OF_CONDUCT.md:64]: perfect. This has all four of our emails.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+## Contacting Us
+
+* To request features or report bugs, please open an
+  [issue](https://github.com/vivarium-collective/vivarium-core/issues).
+* To ask questions or request help, please open a [discussion
+  thread](https://github.com/vivarium-collective/vivarium-core/discussions).
+* To contact the maintainers privately, for example to report a
+  violation of our [code of conduct](CODE_OF_CONDUCT.md) or to notify us
+  of a security issue, you can use the contact information in the
+  [`AUTHORS.md` file](AUTHORS.md).
```
Comment (eagmon) [inline CONTRIBUTING.md:36]: :thumbsup:

**Review by eagmon — APPROVED** [review]
This is looking really good. Thanks @U8NWXD! I'll plan to do an additional read-through of the readme in a separate PR.

**Comment by eagmon** [comment]
@U8NWXD -- I'm redoing much of the readme in #144, including all of @1fish2's suggestions.

File: CODE_OF_CONDUCT.md (hunk 4 above)
Comment (U8NWXD) [reply CODE_OF_CONDUCT.md:122]: It gets rendered as a link, but the target comes from a footnote-like statement elsewhere in the doc. Here's some more info on these kinds of links: https://www.markdownguide.org/basic-syntax#reference-style-links

File: README.md (hunk 3 above)
Comment (U8NWXD) [reply README.md:54]: Yep! fixed

**Pull request merged**